src/twisted/mail/test/test_mailmail.py

```diff
@@ -73,3 +109,271 @@ def test_copyAliasedSender(self):
+
+        argv = ("test_mailmail.py", "invaliduser2@example.com", "-oep")
+        self.patch(sys, 'argv', argv)
+        mailmail.run()
+        self.assertRegex(self.out.getvalue(),
+                         "Illegal UID in \\[useraccess\\] section: "
+                         "invaliduser1")
+        self.assertRegex(self.out.getvalue(),
+                         "Illegal GID in \\[groupaccess\\] section: "
+                         "invalidgid1")
+        self.assertRegex(self.out.getvalue(),
+                         'Illegal entry in \\[identity\\] section: funny')
+
+    if platformType == "win32":
+        test_readInvalidConfig.skip = ("mailmail.run() does not work on win32"
+                                       " due to lack of support for getuid()")
+
+
+    def _loadConfig(self, config):
+        """
+        Read a mailmail configuration file.
+
+        The mailmail script checks the twisted.mail.scripts.mailmail.GLOBAL_CFG
+        variable and then the twisted.mail.scripts.mailmail.LOCAL_CFG
+        variable for the path to its  config file.
+
+        @param config: path to config file
+        @type config: L{str}
+
+        @return: A parsed config.
+        @rtype: L{twisted.mail.scripts.mailmail.Configuration}
+        """
+
+        from twisted.mail.scripts.mailmail import loadConfig
+
+        filename = self.mktemp()
+
+        with open(filename, "w") as f:
+            f.write(config)
+
+        return loadConfig(filename)
+
+
+    def test_loadConfig(self):
+        """
+        L{twisted.mail.scripts.mailmail.loadConfig}
+        parses the config file for mailmail.
+        """
+        config = self._loadConfig("""
+[addresses]
+smarthost=localhost""")
+        self.assertEqual(config.smarthost, "localhost")
+
+        config = self._loadConfig("""
+[addresses]
+default_domain=example.com""")
+        self.assertEqual(config.domain, "example.com")
+
+        config = self._loadConfig("""
+[addresses]
+smarthost=localhost
+default_domain=example.com""")
+        self.assertEqual(config.smarthost, "localhost")
+        self.assertEqual(config.domain, "example.com")
+
+        config = self._loadConfig("""
+[identity]
+host1=invalid
+host2=username:password""")
+        self.assertNotIn("host1", config.identities)
+        self.assertEqual(config.identities["host2"], ["username", "password"])
+
+        config = self._loadConfig("""
+[useraccess]
+allow=invalid1,35
+order=allow""")
+        self.assertEqual(config.allowUIDs, [35])
+
+        config = self._loadConfig("""
+[useraccess]
+deny=35,36
+order=deny""")
+        self.assertEqual(config.denyUIDs, [35, 36])
+
+        config = self._loadConfig("""
+[useraccess]
+allow=35,36
+deny=37,38
+order=deny""")
+        self.assertEqual(config.allowUIDs, [35, 36])
+        self.assertEqual(config.denyUIDs, [37, 38])
+
+        config = self._loadConfig("""
+[groupaccess]
+allow=gid1,41
+order=allow""")
+        self.assertEqual(config.allowGIDs, [41])
```

Comment:
please consider splitting into separate tests with dedicated docstrings.
I am not sure why gid1 is ignored here... I guess because is not a number, but please be explicit in the tests.. as there is no code to test that the test code is valid :)